File: Examples/Ruby/Worksheet/put_work_sheet_background.rb
require 'aspose_cells_cloud'

class Worksheet

  include AsposeCellsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @cells_api = CellsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def put_work_sheet_background
    file_name = "myWorkbook.xlsx"
    upload_file(file_name)

    sheet_name = "Sheet1"
    background_image = File.open("../data/Creative.jpg","r") { |io| io.read }
    response = @cells_api.put_work_sheet_background(file_name, sheet_name, background_image)
  end

end

worksheet = Worksheet.new()
puts worksheet.put_work_sheet_background

File: Examples/PHP/Worksheet/PutWorkSheetBackground.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Cells\CellsApi;
use Aspose\Cells\AsposeApp;

class Worksheet {

	public $cells;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->cells = new CellsApi();
	}

	public function putWorkSheetBackground() {
		// Upload file to Aspose Cloud Storage
		$fileName = "myWorkbook.xlsx";
		Utils::uploadFile($fileName);
		
        $file = realpath(__DIR__ . '/../../..') . '/Data/Creative.jpg';
        $result = $this->cells->PutWorkSheetBackground($name=$fileName, $sheetName="Sheet1", $storage = null, $folder = null, $file);
        print_r($result);
	}
}

$worksheet = new Worksheet();
$worksheet->putWorkSheetBackground();

?>
//ExEnd:
